File: CS202计算机系统基础/37_C是世界上最好的编程语言/Calculatexsinx.py
import math
import time

start = time.perf_counter()

x = 0.0
while x <= 10000.0:
    fx = x * math.sin(x)
    x += math.pow(2, -8)

duration = time.perf_counter() - start
print("Time spent: " + str(duration * 10 ** 3) + " milliseconds.")

File: CS202计算机系统基础/37_C是世界上最好的编程语言/Calculatexsinx.c
#include<math.h>
#include<sys/time.h>
#include<stdio.h>

int main(void){
    struct timeval stop, start;
    gettimeofday(&start, NULL);
    double x = 0.0, fx;
    
    while(x <= 10000.0){
        fx = x * sin(x);
        x += pow(2, -8);
        //printf("%f\n", fx);
    }

    gettimeofday(&stop, NULL);
    double timeInMs = ((stop.tv_sec - start.tv_sec) * 1000000 + stop.tv_usec - start.tv_usec) / 1000.0;
    printf("took %.2f ms\n",  timeInMs);
}